s3pipeline/pipelines.py

```diff
@@ -59,26 +64,31 @@ def open_spider(self, spider):
         """
         # Store timestamp to replace {time} in S3PIPELINE_URL
         self.ts = datetime.utcnow().replace(microsecond=0).isoformat().replace(':', '-')
+        self._spider = spider
+        self._timer = None
 
     def close_spider(self, spider):
         """
         Callback function when spider is closed.
         """
         # Upload remained items to S3.
-        self._upload_chunk(spider)
+        self._upload_chunk()
+        if self._timer is not None:
+            self._timer.cancel()
```

Comment:
Just calling `_timer_cancel()` seems to be better.

```suggestion
        self._timer_cancel()
```

Reply:
Yes I forgot to delete it